### PR TITLE
feat(cli): usage analytics — command lifecycle, MCP, App Created source, Doctor

### PR DESCRIPTION
**Pull request by WcaleNieWolny**
## Summary

Adds the new-events analytics layer on top of the v2 base from #2357. Answers the original product questions:

- **How many use the Capgo MCP?** → `MCP Server Started` + `MCP Tool Invoked` (a single `server.tool` interceptor covers all 24 tools; nothing per-tool to maintain).
- **How many create an app *without* onboarding?** → `App Created` now carries a `source` tag = `cli-direct` / `onboarding` / `mcp`.
- **How often is *any* command run (and its success/failure)?** → universal `CLI Command Invoked / Succeeded / Failed` for **every** command, via Commander `preAction`/`postAction` + the `parseAsync` exit handler. This alone lights up the entire command surface.
- Plus **`Doctor Ran`** (`is_outdated` / `dependency_count` / `outdated_count`) for version-staleness.

### Foundation — `cli/src/analytics/`
- **`track.ts`** — global props (`cli_version`, node/os/arch, `is_ci`, `ci_provider`, `is_tty`, **`invocation_source`**: cli|mcp), opt-out (`CAPGO_DISABLE_TELEMETRY`/`CAPGO_DISABLE_POSTHOG`), v2 `trackEvent` with **best-effort `app_id → owner_org`** resolution from the Capacitor config, a flush registry (events survive process exit), the lifecycle emitters, and the MCP tool wrapper.
- **`error-category.ts`** — PII-free CLI error categorizer (closed enum, never raw text).
- **`org-resolver.ts`** — shared `resolveOwnerOrgId`, extracted from `build/onboarding/analytics.ts` (which now reuses it).

All events flow through the existing `sendEvent` → `/private/events` → PostHog **v2** pipeline (`org_id` + `tracking_version: 2`; the backend derives the actor from the capgkey).

## Privacy
Only flag **names**, closed enums, counts, sizes, durations, and `app_id` are sent — never flag values, paths, tokens, or error text. Honors the opt-out env vars (the new layer; also a clean test kill-switch). Events fire only when an API key is resolvable — local/keyless commands (`doctor`, `key`, `zip`, …) are best-effort (logged-in machines), the accepted unauthenticated gap.

## Test plan
- [x] `bun run build` (tsc typecheck + bundle)
- [x] 6 new unit tests: analytics, error-category, org-resolver, mcp-analytics, app-created-source, doctor-analytics
- [x] `test:mcp`, `test:onboarding-telemetry` (no regression)
- [ ] CI: oxlint, tsgo typecheck, full CLI suite

## Note on the long-tail
The ~19 per-command domain events (Apps/Bundles Listed, Bundle Compatibility Checked, Update Probed, Repo Starred, Encryption Keys, Org Members Listed, …) and existing-event enrichments are **optional follow-on commits to this same PR** — they add per-command *properties* (counts/results), but command *frequency* is already covered for all commands by the universal lifecycle above.

## Summary by CodeRabbit

* **Tests**
  * Added CLI tests for analytics flows (error categorization, org resolution, MCP tooling, app-create source, doctor analytics) and wired them into the aggregated test runner.

* **New Features**
  * Wide telemetry added across CLI and MCP: command lifecycle events, tool invocations, app/credential/bundle/account actions, and doctor dependency metrics.

* **Chores**
  * Improved telemetry reliability: invocation-source handling, error categorization, org-id caching, cancellable requests, and graceful flush/abort of pending analytics.

[Review Change Stack](https://app.coderabbit.ai/change-stack/Cap-go/capgo/pull/2358?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: cli/src/analytics/track.ts [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+  void promise.finally(() => pending.delete(promise))
+}
+
+export async function flushAnalytics(timeoutMs = 2000): Promise<void> {
+  if (pending.size === 0)
+    return
+  await Promise.race([
+    Promise.allSettled([...pending]),
+    new Promise(resolve => setTimeout(resolve, timeoutMs)),
+  ])
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/analytics/track.ts:71]: Fixed in dc3411ea. `flushAnalytics` now aborts all in-flight telemetry (and unref's its own timer) when the 2s window expires, and the abort signal is threaded through `getRemoteConfig` (previously un-timed), `sendEvent`, and the org-resolver query — so their sockets close and the event loop drains promptly. No `exit()` is used, so long-running commands (`mcp`/`debug`) keep their own handles and are unaffected. Added a flush-abort regression test.

File: cli/src/analytics/track.ts [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+        return {}
+      const orgId = await resolveOwnerOrgId(apikey, appId)
+      return { appId, orgId }
+    }
+    catch {
+      return {}
+    }
+  })()
+  return cachedContext
+}
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/analytics/track.ts]: Done in 86f3b4b — cachedContext is now a Map keyed by apikey, so events for different accounts never reuse another resolved org.

File: cli/src/index.ts [hunk 3]
```diff
@@ -1097,7 +1104,9 @@ program.configureOutput({
   },
 })
 
-program.parseAsync().catch(async (error: unknown) => {
+program.parseAsync().then(async () => {
+  await flushAnalytics()
+}).catch(async (error: unknown) => {
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/index.ts]: Done in 86f3b4b — converted the entrypoint to an async IIFE with try/catch; flushAnalytics runs on success and (now) before the help/version early exit.

File: cli/test/test-doctor-analytics.mjs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+assert.equal(tags.outdated_count, 1)
+
+const fresh = computeDoctorAnalyticsTags(
+  { '@capgo/capacitor-updater': '6.2.0' },
+  { '@capgo/capacitor-updater': '6.2.0' },
+)
+assert.equal(fresh.is_outdated, false)
+assert.equal(fresh.outdated_count, 0)
+
+console.log('✅ doctor analytics tags tests passed')
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/test/test-doctor-analytics.mjs:34]: Done in 86f3b4b — added empty-deps and all-outdated edge-case assertions.

File: cli/test/test-doctor-analytics.mjs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+assert.equal(tags.is_outdated, true, 'updater behind latest => outdated')
+assert.equal(tags.dependency_count, 2)
+assert.equal(tags.outdated_count, 1)
+
+const fresh = computeDoctorAnalyticsTags(
+  { '@capgo/capacitor-updater': '6.2.0' },
+  { '@capgo/capacitor-updater': '6.2.0' },
+)
+assert.equal(fresh.is_outdated, false)
+assert.equal(fresh.outdated_count, 0)
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/test/test-doctor-analytics.mjs:21]: Done in 86f3b4b — added the dependency_count assertion to the fresh case.

File: cli/test/test-mcp-analytics.mjs [hunk 6]
```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env node
+import assert from 'node:assert/strict'
+import { flushAnalytics, setInvocationSource, withMcpToolTracking } from '../src/analytics/track.ts'
+
+console.log('🧪 Testing MCP tool tracking...\n')
+
+const originalFetch = globalThis.fetch
+const originalDisable = process.env.CAPGO_DISABLE_TELEMETRY
+const originalToken = process.env.CAPGO_TOKEN
+
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/test/test-mcp-analytics.mjs:11]: Done in 86f3b4b — the test now saves/clears/restores CAPGO_DISABLE_POSTHOG too (same hardening applied to test-analytics).

File: cli/src/build/credentials-command.ts [hunk 7]
```diff
@@ -605,6 +606,8 @@ export async function listCredentialsCommand(options?: { appId?: string, local?:
     log.info(`Local:  ${getLocalCredentialsPath()}`)
     log.info('\n🔒 These credentials are stored locally on your machine only.')
     log.info('   When building, they are sent to Capgo but NEVER stored there.\n')
+
+    void trackEvent({ channel: 'credentials', event: 'Credentials Listed', icon: '📋', tags: { credentials_count: allAppIds.length } })
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/build/credentials-command.ts]: Done in 86f3b4b — credentials_count now uses appsToShow.length (the displayed/filtered count).

File: cli/src/build/credentials-manage.ts [hunk 8]
```diff
@@ -493,6 +494,8 @@ export async function manageCredentialsCommand(options: ManageCredentialsOptions
 
     if (!handedOffToOnboarding)
       pOutro('Done.')
+
+    void trackEvent({ channel: 'credentials', event: 'Credentials Managed', icon: '🗂️', tags: {} })
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/build/credentials-manage.ts]: Done in 86f3b4b — Credentials Managed now only emits inside the !handedOffToOnboarding branch.

File: cli/src/github-command.ts [hunk 9]
```diff
@@ -11,6 +12,7 @@ export function starRepositoryCommand(repository?: string) {
   else {
     log.success(`🙏 Thanks for starring ${fullRepo} 🎉`)
   }
+  void trackEvent({ channel: 'cli-usage', event: 'Repo Starred', icon: '⭐', tags: { repo_count: 1 } })
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/github-command.ts]: Done in 86f3b4b — single-repo event moved into the newly-starred else branch; star-all only emits when starredCount > 0.

File: cli/src/index.ts [hunk 10]
```diff
@@ -1097,37 +1104,47 @@ program.configureOutput({
...
+          })
+        : Promise.resolve(false)
+      // For actual errors, show just the message without the full stack trace
+      if (commanderError.message) {
+        log.error(commanderError.message)
+      }
+      const exitCode = commanderError.exitCode ?? 1
+      if (shouldCapturePosthogException(error))
+        trackCommandFailed(currentCommandPath, { errorCategory: categorizeCliError(error), exitCode })
+      await Promise.all([capturePromise, flushAnalytics()])
```
[the remark this answers is not in the document]
Comment (WcaleNieWolny) [reply cli/src/index.ts]: Done in 088841e — trackCommandFailed now fires unconditionally for every failure (commander usage errors are categorized error_category=commander); only the PostHog exception capture stays gated behind shouldCapturePosthogException.
